src/main/scala/firrtl/annotations/AnnotationYamlProtocol.scala

```diff
@@ -8,30 +8,28 @@ import net.jcazevedo.moultingyaml._
 object AnnotationYamlProtocol extends DefaultYamlProtocol {
   // bottom depends on top
   implicit object AnnotationYamlFormat extends YamlFormat[Annotation] {
-    def write(a: Annotation) =
-      YamlArray(
-        YamlString(a.targetString),
-        YamlString(a.transformClass),
-        YamlString(a.value))
+    def write(a: Annotation) = YamlObject(
+      YamlString("targetString") -> YamlString(a.targetString),
+      YamlString("transformClass") -> YamlString(a.transformClass),
+      YamlString("value") -> YamlString(a.value)
+    )
 
-    def read(value: YamlValue) = {
-      value.asYamlObject.getFields(
+    def read(yamlValue: YamlValue): Annotation = {
+      yamlValue.asYamlObject.getFields(
         YamlString("targetString"),
         YamlString("transformClass"),
         YamlString("value")) match {
-        case Seq(
-          YamlString(targetString),
-          YamlString(transformClass),
-          YamlString(value)) =>
-          new Annotation(toTarget(targetString), Class.forName(transformClass).asInstanceOf[Class[_ <: Transform]], value)
-        case _ => deserializationError("Color expected")
+          case Seq(YamlString(targetString), YamlString(transformClass), YamlString(value)) =>
+              Annotation(
+                toTarget(targetString), Class.forName(transformClass).asInstanceOf[Class[_ <: Transform]], value)
+          case _ => deserializationError("Color expected")
       }
     }
-    def toTarget(string: String) = string.split('.').toSeq match {
+    def toTarget(string: String): Named = string.split('.').toSeq match {
       case Seq(c) => CircuitName(c)
       case Seq(c, m) => ModuleName(m, CircuitName(c))
-      case Nil => error("BAD")
-      case s => ComponentName(s.drop(2).mkString("."), ModuleName(s(1), CircuitName(s(0))))
+      case Nil => sys.error("BAD")
```

Comment:
Will change to do that, but as written it was using Predef.error which is deprecated